### PR TITLE
Test: Set Contact Handle #259

### PR DESCRIPTION
**Pull request by ChrisFernandezVivas**
# Pull Request

- [ x] Closes 259
- [ x] Added tests (if necessary)
- [x ] Run tests
- [ ] Run formatting
- [x ] Commented the code

## Changes description

add a test for a set_contact_handle method, the test check with all authorized users and try with 1 another user, the test always fails because when another user try call the function the test fail

## Current output

Captura de pantalla 2024-11-25 a las 3 45 57 p  m

## Time spent breakdown

5 hours

## Comments

i putted a #[should_panic] so if you remove this the test always fail

### CODE REVIEW

File: contracts/tests/test_fund.cairo [hunk 1]
```diff
@@ -502,3 +502,49 @@ fn test_emit_event_donation_withdraw() {
...
+    let reverted_contact_handle = dispatcher.get_contact_handle();
+    assert(reverted_contact_handle == CONTACT_HANDLE_1(),' revert ');
+
+    // set a no valid address
+    //always should fail
+    start_cheat_caller_address_global(OTHER_USER());
+    dispatcher.set_contact_handle(CONTACT_HANDLE_2());//set
+    let new_contact_handle = dispatcher.get_contact_handle();//get
+    assert(new_contact_handle == CONTACT_HANDLE_1(),'check the function');
+
```
Comment (adrianvrj) [inline contracts/tests/test_fund.cairo:547]: please remove this blank spaces

File: contracts/tests/test_fund.cairo [hunk 2]
```diff
@@ -502,3 +502,49 @@ fn test_emit_event_donation_withdraw() {
...
+#[should_panic(expected: ("You must be an owner or admin to perform this action",))]
+fn test_set_contact_handle(){
+    let contract_address = _setup_();
+    let dispatcher = IFundDispatcher { contract_address };
+    let contact_handle = dispatcher.get_contact_handle();
+    assert(contact_handle == CONTACT_HANDLE_1(), 'Invalid contact handle');
+
+    //owner
+    start_cheat_caller_address_global(OWNER());
+    dispatcher.set_contact_handle(CONTACT_HANDLE_2());//set
```
Comment (adrianvrj) [inline contracts/tests/test_fund.cairo:516]: remove this ```set/get``` comments please

**Review by adrianvrj — CHANGES_REQUESTED** [review]
just a couple changes from my side, did you run scarb fmt?

File: contracts/tests/test_fund.cairo [hunk 3]
```diff
@@ -502,3 +502,49 @@ fn test_emit_event_donation_withdraw() {
         )
     ]);
 }
+
+#[test]
+#[should_panic(expected: ("You must be an owner or admin to perform this action",))]
+fn test_set_contact_handle(){
```
Comment (EmmanuelAR) [inline contracts/tests/test_fund.cairo:508]: Please divided the test in success case and error case

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

File: contracts/tests/test_fund.cairo (hunk 2 above)
Comment (ChrisFernandezVivas) [reply contracts/tests/test_fund.cairo:516]: okey perfect I will do

File: contracts/tests/test_fund.cairo (hunk 1 above)
Comment (ChrisFernandezVivas) [reply contracts/tests/test_fund.cairo:547]: good

File: contracts/tests/test_fund.cairo (hunk 3 above)
Comment (ChrisFernandezVivas) [reply contracts/tests/test_fund.cairo:508]: okey thank you for your feedback

**Review by EmmanuelAR — APPROVED** [review]
lgtm

**Review by adrianvrj — APPROVED** [review]

**Pull request merged**